File: ps/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int i,j=0, res = 0;
        for(i=0;i<prices.size();i++) {
            if (prices[i] < prices[j]) j = i;
            res = max(res, prices[i] - prices[j]);
        }
        return res;
    }
};

File: ps/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxGap = 0
        stack = []
        for price in prices:
            if not stack:
                stack.append(price)
                continue
            if stack[-1] < price:
                maxGap = max(maxGap, price - stack[-1])
            else:
                stack.pop()
                stack.append(price)

        return maxGap